vignettes/example_analysis_coxreg.Rmd

```diff
@@ -0,0 +1,380 @@
+
+Using these arguments, the `cached_model` function first checks if a model for the given covariate `cov` is already stored in the caching environment `cache_env`. If so, then this model is retrieved and returned by `cached_model`. If not, the model must be constructed. This is done by first constructing the model formula, `model_form`, starting with only the treatment effect (`ARM`) and adding a covariate effect if one is currently being analyzed. Then a Cox regression model is fit using `df` and the model formula, and this model is both returned and stored in the caching environment object as `cache_env[[cov]]`.
+
+```{r}
+cached_model <- function(df, cov, cache_env) {
+  ## Check if a model already exists for
+  ## `cov` in the caching environment
+  if (!is.null(cache_env[[cov]])) {
+    ## If model already exists, retrieve it from cache_env
+    model <- cache_env[[cov]]
+  } else {
+    ## Build model formula
+    model_form <- paste0("survival::Surv(AVAL, EVENT) ~ ARM")
+    if (length(cov) > 0) {
+      model_form <- paste(c(model_form, cov), collapse = " * ")
+    } else {
+      cov <- "ARM"
+    }
+    ## Calculate Cox regression model
+    model <- survival::coxph(
+      formula = stats::as.formula(model_form),
+      data = df,
+      ties = "exact"
+    )
+    ## Store model in the caching environment
+    cache_env[[cov]] <- model
+  }
+  model
+}
+```
+
+## Creating the Analysis Function: `a_cox_summary`
+
+With our data prepared and helper function created, we can proceed to construct our analysis function `a_cox_summary`, which will be used to populate all of the rows in our table. In order to be used to generate both data rows (for interaction effects) and content rows (for main effects), we must create a function that can be used as both `afun` in `analyze` and `cfun` in `summarize_row_groups`. Therefore, our function must accept either `x` or `df` as its first parameter and also accept the `labelstr` parameter.
+
+The arguments of our analysis function will be as follows:
+
+-   `df` - a `data.frame` of the full dataset required to fit the Cox regression model.
+-   `labelstr` - the `string` label for the variable being analyzed in the current row/column split context.
+-   `.spl_context` - a `data.frame` containing the `value` column which is used by this analysis function to determine the name of the variable/covariate in the current split. For more details on the information stored by `.spl_context` see `?analyze`.
+-   `stat` and `format` - `string`s that indicate which statistic column we are currently in and what format should be applied to print the statistic.
+-   `cache_env` - an `environment` object that can be used to store cached models so that we can prevent repeatedly fitting the same model. Instead, each model will be generated once per covariate and then reused.
```

Comment:
say explicitly that this will be passed directly down to the `cached_model` helper we defined